smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/HostLabelTraitValidator.java

```diff
@@ -154,4 +159,14 @@
         }
         return Optional.empty();
     }
+
+    private Optional<ValidationEvent> validateTrailingPeriod(OperationShape operation, EndpointTrait trait) {
+        SmithyPattern hostPrefix = trait.getHostPrefix();
+        if (!hostPrefix.toString().endsWith(".") && !hostPrefix.getLabels().isEmpty()) {
+            String message = "`endpoint` trait hostPrefix contains host labels and does not end in a period (`.`). "
+                    + "This can result in clients inadvertently sending data to domains you do not control.";
+            return Optional.of(warning(operation, trait, message));
```

Comment:
I am also open to changing this to a DANGER. Right now 8 AWS services use a `hostPrefix` and none of them would run afoul of this validator.

Reply:
DANGER (or an outright failure) I think is appropriate..

Reply:
+1 on making this a DANGER, given it's specific to the label-having prefixes.

Reply:
done